File: be/SGBD.Domain/Entities/Articole.cs
﻿namespace SGBD.Domain.Entities
{
    public partial class Articole : BaseEntity
    {
        public decimal? IdComenzi { get; set; }
        public decimal? IdStoc { get; set; }
        public decimal? Cantitate { get; set; }
        public decimal? PretTotal { get; set; }

        public virtual Comenzi? Comenzi { get; set; }
        public virtual Stoc? Stoc { get; set; }
    }
}

File: fe/lib/models/articole.dart
class Articole {
  double? id;
  double? idComenzi;
  double? idStoc;
  double? cantitate;
  double? pretTotal;

  Articole({
    this.id,
    this.idComenzi,
    this.idStoc,
    this.cantitate,
    this.pretTotal,
  });

  factory Articole.fromJson(Map<String, dynamic> json) {
    return Articole(
      id: json['id']?.toDouble(),
      idComenzi: json['idComenzi']?.toDouble(),
      idStoc: json['idStoc']?.toDouble(),
      cantitate: json['cantitate']?.toDouble(),
      pretTotal: json['pretTotal']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'idComenzi': idComenzi,
      'idStoc': idStoc,
      'cantitate': cantitate,
      'pretTotal': pretTotal,
    };
  }
}
